### PR TITLE
[microNPU] Merge LUT activation with binary elementwise operation

### PR DESCRIPTION
**Pull request by Alexey-Yazev**
Add binary elementwise operator to OptimizeLUTs pass to merge LUT activation with elementwise operation. This PR is preparation for implementation of SoftMax operation.

cc @leandron, @ekalda, @lhutton1

### CODE REVIEW

**Review by lhutton1 — APPROVED** [review]
Thanks @Alexey-Yazev, looks sensible to me, I just had a small nit from a previous issue with constant data types that should now be fixed

File: tests/python/contrib/test_ethosu/infra.py [hunk 1]
```diff
@@ -694,11 +694,12 @@ def make_ethosu_binary_elementwise(
     use_rescale: bool = False,
     rescale_scale: int = 0,
     rescale_shift: int = 0,
+    lut=relay.const([], dtype="int8"),
```
[the remark this answers is not in the document]
Comment (ekalda) [reply tests/python/contrib/test_ethosu/infra.py:697]: I can't remember... IIRC we can only have LUT in `int8` or `int16`, so theoretically that const there as a part of infra could also be `int16`, but in practice we don't have any `int16` LUTs implemented, so not sure how much point there is testing it... I'm fine either way :)

**Review by ekalda — APPROVED** [review]
Thanks @Alexey-Yazev, LGTM! When I initially implemented the LUT support, there was some binary elementwise issue I couldn't figure out, I suppose your previous fixes to binary elementwise have also removed that underlying issue... Great to see this fixed!

File: tests/python/contrib/test_ethosu/infra.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (lhutton1) [reply tests/python/contrib/test_ethosu/infra.py:697]: Ah okay, let's leave it for now then

**Comment by lhutton1** [comment]
Thanks @Alexey-Yazev, @ekalda!

**Pull request merged**